code/modules/reagents/Chemistry-Goon-Medicine.dm

```diff
@@ -0,0 +1,335 @@
+	color = "#C8A5DC" // rgb: 200, 165, 220
+
+datum/reagent/charcoal/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustToxLoss(-2*REM)
+	..()
+	return
+
+/datum/chemical_reaction/charcoal
+	name = "Charcoal"
+	id = "charcoal"
+	result = "charcoal"
+	required_reagents = list("ash" = 1, "sodiumchloride" = 1)
+	result_amount = 2
+	mix_message = "The mixture yields a fine black powder."
+	required_temp = 380
+
+/datum/chemical_reaction/silver_sulfadiazine
+	name = "Silver Sulfadiazine"
+	id = "silver_sulfadiazine"
+	result = "silver_sulfadiazine"
+	required_reagents = list("ammonia" = 1, "silver" = 1, "sulfur" = 1, "oxygen" = 1, "chlorine" = 1)
+	result_amount = 5
+
+/datum/chemical_reaction/salglu_solution
+	name = "Saline-Glucose Solution"
+	id = "salglu_solution"
+	result = "salglu_solution"
+	required_reagents = list("sodiumchloride" = 1, "water" = 1, "sugar" = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/synthflesh
+	name = "Synthflesh"
+	id = "synthflesh"
+	result = "synthflesh"
+	required_reagents = list("blood" = 1, "carbon" = 1, "styptic_powder" = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/styptic_powder
+	name = "Styptic Powder"
+	id = "styptic_powder"
+	result = "styptic_powder"
+	required_reagents = list("aluminium" = 1, "hydrogen" = 1, "oxygen" = 1, "sacid" = 1)
+	result_amount = 4
+	mix_message = "The solution yields an astringent powder."
+
+datum/reagent/omnizine
+	name = "Omnizine"
+	id = "omnizine"
+	description = "Heals one each of OXY, TOX, BRUTE and BURN per cycle."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 0.2
+
+datum/reagent/omnizine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustToxLoss(-1*REM)
+	M.adjustOxyLoss(-1*REM)
+	M.adjustBruteLoss(-1*REM)
+	M.adjustFireLoss(-1*REM)
+	..()
+	return
+
+datum/reagent/potass_iodide
+	name = "Potassium Iodide"
+	id = "potass_iodide"
+	description = "80% chance of removing 1 RAD. Radiation is cumulative and causes tox+burn."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+
+datum/reagent/potass_iodide/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.radiation > 0)
+		if(prob(80))
+			M.radiation--
+	if(M.radiation < 0)
+		M.radiation = 0
+	..()
+	return
+
+/datum/chemical_reaction/potass_iodide
+	name = "Potassium Iodide"
+	id = "potass_iodide"
+	result = "potass_iodide"
+	required_reagents = list("potassium" = 1, "iodine" = 1)
+	result_amount = 2
+
+datum/reagent/pen_acid
+	name = "Pentetic Acid"
+	id = "pen_acid"
+	description = "80% chance of removing 1 RAD. Radiation is cumulative and causes tox+burn."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+
+datum/reagent/pen_acid/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.radiation > 0)
+		M.radiation -= 7
+	M.adjustToxLoss(-4*REM)
+	if(prob(33))
+		M.adjustBruteLoss(1*REM)
+	if(M.radiation < 0)
+		M.radiation = 0
+	..()
+	return
+
```

Comment:
chemical purging was broken and has been scrapped for the current build, it will be readded if I can get a working version of it that doesn't require intense snowflake.